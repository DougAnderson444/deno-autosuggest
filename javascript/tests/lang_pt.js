const {LucidSuggest} = require('../pt')


describe('Portuguese language', () => {
    const records = [
        {id: 10, title: 'Pilhas Alcalinas AA — Embalagem com 12'},
        {id: 20, title: 'Cabo Lightning para USB-A'},
        {id: 30, title: 'Escova de dentes elétrica'},
        {id: 40, title: 'Sacos de armazenamento a vácuo'},
    ]

    const suggest = new LucidSuggest()
    suggest.setRecords(records)

    test('Empty input', async () => {
        const hits = await suggest.search('')
        expect(hits).toMatchSnapshot()
    })

    test('Equality', async () => {
        const hits = await suggest.search('escova de dentes eletrica')
        expect(hits).toMatchSnapshot()
    })

    test('Stemming', async () => {
        const hits = await suggest.search('alcalino')
        expect(hits).toMatchSnapshot()
    })

    test('Partiles', async () => {
        const hits = await suggest.search('de')
        expect(hits).toMatchSnapshot()
    })

    test('Normalization', async () => {
        const hits = await suggest.search('electric')
        expect(hits).toMatchSnapshot()
    })
})