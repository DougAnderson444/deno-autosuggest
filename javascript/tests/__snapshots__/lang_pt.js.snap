// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portuguese language Empty input 1`] = `
Array [
  Object {
    "id": 10,
    "rating": 4,
    "title": "Pilhas Alcalinas AA — Embalagem com 12",
  },
  Object {
    "id": 20,
    "rating": 3,
    "title": "Cabo Lightning para USB-A",
  },
  Object {
    "id": 30,
    "rating": 2,
    "title": "Escova de dentes elétrica",
  },
  Object {
    "id": 40,
    "rating": 1,
    "title": "Sacos de armazenamento a vácuo",
  },
]
`;

exports[`Portuguese language Equality 1`] = `
Array [
  Object {
    "id": 30,
    "rating": 2,
    "title": "[Escova] [de] [dentes] [elétrica]",
  },
  Object {
    "id": 40,
    "rating": 1,
    "title": "Sacos [de] armazenamento a vácuo",
  },
]
`;

exports[`Portuguese language Normalization 1`] = `
Array [
  Object {
    "id": 30,
    "rating": 2,
    "title": "Escova de dentes [elétric]a",
  },
]
`;

exports[`Portuguese language Partiles 1`] = `
Array [
  Object {
    "id": 30,
    "rating": 2,
    "title": "Escova de [de]ntes elétrica",
  },
  Object {
    "id": 40,
    "rating": 1,
    "title": "Sacos [de] armazenamento a vácuo",
  },
]
`;

exports[`Portuguese language Stemming 1`] = `
Array [
  Object {
    "id": 10,
    "rating": 4,
    "title": "Pilhas [Alcalina]s AA — Embalagem com 12",
  },
]
`;
